sapp/tests/warning_messages_test.py

```diff
@@ -0,0 +1,84 @@
+import json
+import pathlib
+from unittest import TestCase
+
+from sqlalchemy import Table
+
+from ..db import DB, DBType
+from ..models import WarningMessage, create as create_model
+from ..warning_messages import update_warning_messages, upsert_entry
+from .fake_object_generator import FakeObjectGenerator
+
+test_metadata = {
+    "codes": {
+        "1001": "Updated warning message from test",
+        "1002": "Updated warning message from test 2",
+    },
+    "codes2": {
+        "2001": "Updated warning message from another test",
+        "2002": "Updated warning message from another test 2",
+    },
+}
+
+
+class WarningMessagesTest(TestCase):
+    def setUp(self) -> None:
+        self.db = DB(DBType.MEMORY)
+        create_model(self.db)
+        self.fakes = FakeObjectGenerator()
+
+        with open("sample_metadata.json", "w") as outfile:
```

Comment:
The `tempfile` library is slightly more suited for this than rolling your own logid to set up and tear down a test file:
https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile

Reply:
Additionally, this file is currently only used in `test_update_warning_messages`, so there's no need for a dedicated setup/teardown method. The file creation can occur in `test_update_warning_messages` directly

Reply:
done